sdk/core/azure-core/test/ut/pipeline.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "gtest/gtest.h"
+#include <azure/core/http/pipeline.hpp>
+#include <azure/core/http/policy.hpp>
+
+#include <vector>
+
+TEST(Logging, createPipeline)
+{
+  // Construct pipeline without exception
+  std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  policies.push_back(std::make_unique<Azure::Core::Http::TelemetryPolicy>("test", "test"));
+
+  EXPECT_NO_THROW(Azure::Core::Http::HttpPipeline pipeline(policies));
+}
+
+TEST(Logging, createEmptyPipeline)
+{
+  // throw invalid arg for empty policies
+  std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  EXPECT_THROW(Azure::Core::Http::HttpPipeline pipeline(policies), std::invalid_argument);
```

Comment:
here @ahsonkhan , this is the tests for empty policies

Reply:
Then, where's the test for:
https://github.com/Azure/azure-sdk-for-cpp/blob/5dae9451a31cdc58966370bd61543b25e3431ac3/sdk/core/azure-core/inc/azure/core/http/pipeline.hpp#L43-L48

We have two APIs with this new check. Are both tested? I only see one `EXPECT_THROW` in the tests.

Reply:
got it.
Actually I am glad you mentioned because I noticed a test case was wrong
I have fixed it and added the tests for the double ref empty vector.

Reply:
btw, this tests covers the fisrt constructor (&)
below tests covers the move constructor (&&)